TwitchDownloaderCore/TwitchHelper.cs

```diff
@@ -189,6 +190,10 @@ public static async Task<EmoteResponse> GetThirdPartyEmoteData(string streamerId
                     JObject emoteData = (JObject)stvEmote["data"];
                     JObject emoteHost = (JObject)emoteData["host"];
                     JArray emoteFiles = (JArray)emoteHost["files"];
+                    if (emoteFiles.Count == 0) // Sometimes there are no hosted files for the emote
+                    {
+                        continue;
+                    }
```

Comment:
This fixes the 404s for me.
Maybe files that are being processed have this empty?

Reply:
Occasionally a streamer's 7TV response contains an emote that does not have any hosted files for whatever reason. Previously my 7tv v3 implementation only used the `files` section to determine format and then builds the final URL from several other fields because the `files` section only contains a part that also overlaps with another field that is always required to build the final URL, so I took the shortcut ignoring the possibility of an empty `files` section